core/src/main/java/hudson/model/UpdateCenter.java

```diff
@@ -1122,30 +1119,29 @@ public File download(DownloadJob job, URL src) throws IOException {
                 con.setReadTimeout(PLUGIN_DOWNLOAD_READ_TIMEOUT);
                 
                 int total = con.getContentLength();
-                in = new CountingInputStream(con.getInputStream());
                 byte[] buf = new byte[8192];
                 int len;
 
                 File dst = job.getDestination();
                 File tmp = new File(dst.getPath()+".tmp");
-                out = Files.newOutputStream(tmp.toPath());
-                if (sha1 != null) {
-                    out = new DigestOutputStream(out, sha1);
-                }
 
                 LOGGER.info("Downloading "+job.getName());
                 Thread t = Thread.currentThread();
                 String oldName = t.getName();
                 t.setName(oldName + ": " + src);
                 try {
-                    while((len=in.read(buf))>=0) {
-                        out.write(buf,0,len);
-                        job.status = job.new Installing(total==-1 ? -1 : in.getCount()*100/total);
+                    try (OutputStream _out = Files.newOutputStream(tmp.toPath());
```

Comment:
Ah, I was wrong [according to this](http://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html):

> **Note**: A `try`-with-resources statement can have `catch` and `finally` blocks just like an ordinary `try` statement. In a `try`-with-resources statement, any `catch` or `finally` block is run after the resources declared have been closed.